java-checks/src/main/java/org/sonar/java/checks/ValueBasedObjectsShouldNotBeSerializedCheck.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.ValueBasedUtils;
+import org.sonar.java.checks.serialization.SerializableContract;
+import org.sonar.java.model.ModifiersUtils;
+import org.sonar.java.resolve.JavaType;
+import org.sonar.java.resolve.ParametrizedTypeJavaType;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Modifier;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+@Rule(key = "S3437")
+public class ValueBasedObjectsShouldNotBeSerializedCheck extends IssuableSubscriptionVisitor {
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.CLASS, Tree.Kind.ENUM, Tree.Kind.ANNOTATION_TYPE);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    String message = "Make this value-based field transient so it is not included in the serialization of this class.";
```

Comment:
Not inlined, because used twice in the method.
The right scope is the method, no need to declare it a member. Do you want
a "final" modifier?

On Tue, Jul 19, 2016 at 10:57 AM, benzonico <notifications@github.com>
wrote:

> In
> java-checks/src/main/java/org/sonar/java/checks/ValueBasedObjectsShouldNotBeSerializedCheck.java
> <https://github.com/SonarSource/sonar-java/pull/930#discussion_r71297794>:
>
> > +import org.sonar.plugins.java.api.tree.Modifier;
> > +import org.sonar.plugins.java.api.tree.Tree;
> > +import org.sonar.plugins.java.api.tree.Tree.Kind;
> > +import org.sonar.plugins.java.api.tree.VariableTree;
> > +
> > +@Rule(key = "S3437")
> > +public class ValueBasedObjectsShouldNotBeSerializedCheck extends IssuableSubscriptionVisitor {
> > +
> > +  @Override
> > +  public List<Kind> nodesToVisit() {
> > +    return ImmutableList.of(Tree.Kind.CLASS, Tree.Kind.ENUM, Tree.Kind.ANNOTATION_TYPE);
> > +  }
> > +
> > +  @Override
> > +  public void visitNode(Tree tree) {
> > +    String message = "Make this value-based field transient so it is not included in the serialization of this class.";
>
> Message should either be inlined or a constant. but definitely not a local
> var.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SonarSource/sonar-java/pull/930/files/e43a80e0646da5bcefbcd52a60dca56412079a85#r71297794>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASSFNnqKribCO1DhfA9m_bIYc_xtds6Eks5qXJGAgaJpZM4JOqIZ>
> .
>